internal/provider/zip_archiver.go

```diff
@@ -137,11 +144,29 @@ func (a *ZipArchiver) ArchiveDir(indirname string, excludes []string) error {
 			return err
 		}
 
+		if info.Mode()&os.ModeSymlink == os.ModeSymlink {
+			realPath, err := readSymlinkRecursive(path, maxRecursion)
```

Comment:
I commented above about the `maxRecursion` variable and decided to do a little more digging and found that the standard library has a function [`filepath.EvalSymlinks`](https://pkg.go.dev/path/filepath#EvalSymlinks) that looks very similar to this `readSymlinkRecursive` function (with some additional OS specific edge-cases to boot).

We may want to consider using this standard library function if it is providing the same functionality. 

Fun fact, [the magic number in the standard library is 255 symlinks!](https://github.com/golang/go/blob/eee6f2b921e13d10faff6d45dc099c3a45d389b9/src/path/filepath/symlink.go#L99) 😆

Reply:
Sweet! Nice find, thanks. I've refactored to switch to using `filepath.EvalSymlinks`. Looks like the functionality meets our requirements and it handles symlinks to absolute paths correctly out of the box. 👍